node/node.go

```diff
@@ -294,6 +295,13 @@ func NewNode(config *cfg.Config,
 	sw.AddReactor("CONSENSUS", consensusReactor)
 	sw.AddReactor("EVIDENCE", evidenceReactor)
 
+	// Generate node PrivKey
+	nodeKey, err := p2p.LoadOrGenNodeKey(config.NodeKeyFile())
```

Comment:
this needs to be moved into `DefaultNewNode`

```
nodeKey, err := p2p.LoadOrGenNodeKey(config.NodeKeyFile())
if err != nil {
  return nil, err 
}
NewNode(config,
		privval.LoadOrGenFilePV(config.PrivValidatorFile())
                 nodeKey,
```

log statement should stay here

Reply:
Roger that!